app/pages/debug/ListingDemoPage.tsx

```diff
@@ -24,29 +28,41 @@ export const simpleScheduleLookup = {
   random7: [[1000, 2300], [0, 2359], [0, 2359], [0, 2359], [0, 2359], [0, 2359], null],
   random8: [[[2330, 1200], [1800, 2000], [2200, 2359]], null, [0, 2359], [0, 2359], [0, 2359], [0, 2359], [900, 2200]],
   closes_tomorrow: [[1800, 900], [0, 2359], [0, 2359], [0, 2359], [0, 2359], [0, 2359], [[0, 900], [1100, 1800], [1900, 2300]]],
-};
+} as const;
 /* eslint-enable max-len */
 
 const simpleSchedules = Object.values(simpleScheduleLookup);
 
-export const createScheduleFromShorthand = shedule_shorthand => {
+/** Return true if x is an Interval[]; false if it is only an Interval.
+ *
+ * This is a user-defined type guard, since TypeScript is unable to infer the
+ * correct type normally.
+ *
+ * For more information on user-defined type guards, see:
+ * https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates
+ */
+function isIntervalArray(x: Interval | readonly Interval[]): x is readonly Interval[] {
```

Comment:
I still don't understand why the return type here is not a Boolean.

Reply:
It's just special syntax in TypeScript for user-defined type guards. I linked to the TypeScript documentation on user-defined type guards in the comment for this function. Is there a particular part of that documentation that you wanted a more detailed explanation for?

Reply:
Got it. I saw a part of the documentation but there are different kinds of narrowing.